packages/types/src/client.ts

```diff
@@ -23,30 +22,28 @@ export declare namespace ClientTypes {
     | "request"
     | "event";
 
+  interface BaseEventArgs<T> {
+    id: number;
+    topic: string;
+    params: T;
+  }
+
   interface EventArguments {
-    session_proposal: JsonRpcRequest<ProposalTypes.Struct>;
-    session_update: JsonRpcRequest<{ topic: string; namespaces: SessionTypes.Namespaces }>;
-    session_extend: JsonRpcRequest<{ topic: string }>;
-    session_ping: JsonRpcRequest<{ topic: string }>;
-    pairing_ping: JsonRpcRequest<{ topic: string }>;
-    session_delete: JsonRpcRequest<{ topic: string }>;
-    pairing_delete: JsonRpcRequest<{ topic: string }>;
+    session_proposal: Omit<BaseEventArgs<ProposalTypes.Struct>, "topic">;
+    session_update: BaseEventArgs<{ namespaces: SessionTypes.Namespaces }>;
+    session_extend: Omit<BaseEventArgs<unknown>, "params">;
```

Comment:
Shouldn't we be using `<any>` rather than `<unknown>` here if we want `T` to be truly generic?

Reply:
I personally try to avoid any whenever possible. In this case is doesn't matter as props are omitted. I think this is better than doing `params?: T` as with that one, whenever props are actually present users will have to add `if (props)`.

Reply:
Ah yeah in this case it's the omitted key anyways so doesn't matter 👍 

I agree with avoiding `any` generally, except with generics where you just want a truly transparent type/let the end user assign (literally) anything they want. I suspect with `unknown` the TS compiler would ask you to cast from `unknown`, but I'd have to test it to actually be sure.